### ISSUE TITLE
Update extension input to support error

### ISSUE THREAD

**Issue by jeluard** [status-im/status-react#7104]
### User Story

As a developer, I want to have the ability to show that an input value is incorrect. Add an `error` property that when provided is displayed inline.

Design: 
* https://www.figma.com/file/cb4p8AxLtTF3q1L6JYDnKN15/Index?node-id=858%3A64
* https://www.figma.com/file/cb4p8AxLtTF3q1L6JYDnKN15/Index?node-id=858%3A144

`error` would be an extra property added to the extension input component. It could reuse what has been done for the regular input component.

### Description

*Type*: Feature

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 80.0 DAI (80.0 USD @ $1.0/DAI)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/7104/2020).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/status-im/status-react/7104/2020).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $43,768.14 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by mukulverm4** [comment]
Hey @jeluard, the `error` property should be for the react native front-end, right?

**Comment by jeluard** [comment]
`error` would be an extra property added to the extension [input](https://github.com/status-im/status-react/blob/develop/src/status_im/extensions/core.cljs#L319) component. It could reuse what has been done for the regular [input component](https://github.com/status-im/status-react/blob/develop/src/status_im/ui/components/text_input/view.cljs#L26).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 5 months, 3 weeks from now.
Please review their action plans below:


**1) [trantho3910](https://gitcoin.co/profile/trantho3910) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/status-im/status-react/7104/2020?mutate_worker_action=approve&worker=trantho3910) | [reject worker](https://gitcoin.co/issue/status-im/status-react/7104/2020?mutate_worker_action=reject&worker=trantho3910))_.**

oh ahihi  oh ahihi oh ahihi oh ahihi oh ahihi oh ahihi oh ahihi oh ahihi oh ahihi oh ahihi oh ahihi

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/7104/2020).

**Comment by mukulverm4** [comment]
Hey @jeluard, I'm getting this `Error: File not found: index.android.js in any of the project roots` while building. I found an issue on this in the repository that was closed without any resolution. Please help or let me know if I should raise an issue.

**Comment by jeluard** [comment]
@mukulverm4 Looks like you are missing some `react-native` process running? Make sure you follow our docs and don't get any unexpected error message.

**Comment by mukulverm4** [comment]
I tried again and got the following while make setup in Ubuntu
```
Task :react-native-android:androidJavadoc 
/home/hotfudge/Projects/status-react/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/ModuleSpec.java:10: error: package com.facebook.react.common.build does not exist
import com.facebook.react.common.build.ReactBuildConfig;
                                      ^
/home/hotfudge/Projects/status-react/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIViewOperationQueue.java:11: error: package com.facebook.common.logging does not exist
import com.facebook.common.logging.FLog;
                                  ^
...

100 warnings
```

**Comment by jeluard** [comment]
@mukulverm4 Never seen this one. There might be some other errors before this one?

**Comment by mukulverm4** [comment]
No, no error before this.

**Comment by mukulverm4** [comment]
@jeluard I am still unable to build status in Ubuntu and I found out this same [issue](https://github.com/status-im/status-react/issues/5636). Can you confirm that it builds on Ubuntu?

Also, is there any way to build on Windows?

**Comment by jeluard** [comment]
@mukulverm4 Yes I am using ubuntu myself. Not sure about windows.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@mukulverm4 due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/status-im/status-react/7104/2020) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/status-im/status-react/7104/2020?snooze=1) | [3 days](https://gitcoin.co/issue/status-im/status-react/7104/2020?snooze=3) | [5 days](https://gitcoin.co/issue/status-im/status-react/7104/2020?snooze=5) | [10 days](https://gitcoin.co/issue/status-im/status-react/7104/2020?snooze=10) | [100 days](https://gitcoin.co/issue/status-im/status-react/7104/2020?snooze=100)

**Comment by mukulverm4** [comment]
Hey @jeluard, even after many attempts I haven't been able to setup the environment. I am gonna stop working on this now. I am sorry to have wasted your time.

**Comment by StatusSceptre** [comment]
@mukulverm4 This has been far from a waste of time, we're glad to hear your feedback and we are aware of the difficulty some users find setting up our build setup. The good news is there is a great amount of work going on towards making the build process simpler for newcomer developers. Would love it if we could nail down what the problem was and hopefully allow and encourage you to contribute.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 4 months, 3 weeks from now.
Please review their action plans below:


**1) [pbteja1998](https://gitcoin.co/profile/pbteja1998) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/status-im/status-react/7104/2020?mutate_worker_action=approve&worker=pbteja1998) | [reject worker](https://gitcoin.co/issue/status-im/status-react/7104/2020?mutate_worker_action=reject&worker=pbteja1998))_.**

I want to work on this issue. I cannot provide an action plan right away. I have to go through the codebase first.
I will start the work as soon as I get approved.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/status-im/status-react/7104/2020).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 80.0 DAI  (80.0 USD @ $1.0/DAI) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $51,626.08 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by alexanmtz** [comment]
As agreed with @jeluard on #7105 , I will work on this ✋

**Comment by alexanmtz** [comment]
@jeluard I have the project running on emulator (IOS), but there's just the initial screen and the log:

```
Metro Bundler ready.

Loading dependency graph, done.
(node:26418) UnhandledPromiseRejectionWarning: Error: File not found: index.ios.js in any of the project roots (/Users/alexandremagno/projects/status-react)
    at getAbsolutePath (/Users/alexandremagno/projects/status-react/node_modules/metro/src/lib/getAbsolutePath.js:32:9)
    at Server._getOptionsFromUrl (/Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:1060:31)
    at Server._prepareDeltaBundler (/Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:644:26)
    at /Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:764:15
    at Generator.next (<anonymous>)
    at step (/Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:11:1077)
    at /Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:11:1307
    at new Promise (<anonymous>)
    at /Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:11:988
    at Server._processBundleRequest (/Users/alexandremagno/projects/status-react/node_modules/metro/src/Server.js:843:32)
(node:26418) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:26418) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

It's possible to provide the testflight invite for me to see the app and understand better how it works?

**Comment by jeluard** [comment]
Do you have `react-native start` running in a separate terminal?

**Comment by alexanmtz** [comment]
I tried this as well @jeluard , and and executed `make run-ios`

**Comment by alexanmtz** [comment]
So @jeluard , this should be enough to run a app without issues?

**Comment by jeluard** [comment]
@alexanmtz As long as you follow the steps in the docs you should be good yes.

**Comment by alexanmtz** [comment]
Yeah @jeluard , I followed all the steps and still having this error on node. I will try to figure out why this is happening.

**Comment by jeluard** [comment]
@alexanmtz Did you try with recent develop? It should be simpler now

**Comment by alexanmtz** [comment]
Hello @jeluard , I was stuck on this, I will try again today about the progress

**Comment by StatusSceptre** [comment]
@jeluard is this no longer a bounty?  It was cancelled previously.  I'd like to remove it from the board if so.

**Comment by jeluard** [comment]
Feel free to remove it.

**Comment by alexanmtz** [comment]
Hello @jeluard , I could run the project successfully 🚀 😄 

I have the following screen with an input which triggers an error as described here:
![screen-changed](https://user-images.githubusercontent.com/88840/56354970-6875f500-61d5-11e9-8854-9301fc4d5c4e.png)

So this issue is solved and now we need to extend the input to support numerical range in #7105 ?

**Comment by jeluard** [comment]
Great news!

But no it’s not fixed. The issue is about implementing this in extensions.
You have more details in the original issue.

**Comment by alexanmtz** [comment]
I see @jeluard , I'm working on it.

**Comment by alexanmtz** [comment]
Hello @jeluard , I'm having a great progress on this issue and this is a really cool project. I'm still newbie on closure.js, but I can see how the architecture works.

After I changed the code locally and I added the `error` property, I would like to test, and I created an extension like this:

```
{meta {:name          "Test extension example"
       :description   "just to test the input error example"
       :documentation ""}

 views/hello
 [input {:style {:margin 10} :placeholder "Test" :error "test error"}]
 
 hooks/wallet.settings.hello
 {:label   "Test wallet settings"
  :view    [hello]}}
```

And locally I activated this extension and I'm not sure if my changes is reflecting on the extension to trigger the error locally. So this is the right way to test my changes locally? I can count with the package manager and update on the app, or I need to rebuild in this case? 

Thanks!
